lms/djangoapps/grades/tests/test_transformer.py

```diff
@@ -102,6 +102,98 @@ def build_course_with_problems(self, data='<problem></problem>', metadata=None):
+            Vertical 1 contains 1 vertical (11)
+            Vertical 2 contains no children
+            Vertical 3 contains no children
+            Vertical 11 contains 1 problem (aa) and vertical 2
+            Problem b contains no children
+        """
+        return self.build_course([
+            {
+                u'org': u'GradesTestOrg',
+                u'course': u'GB101',
+                u'run': u'cannonball',
+                u'metadata': {u'format': u'homework'},
+                u'#type': u'course',
+                u'#ref': u'course',
+                u'#children': [
+                    {
+                        u'#type': u'chapter',
+                        u'#ref': u'chapter',
+                        u'#children': [
+                            {
+                                u'#type': u'sequential',
+                                u'#ref': 'sub_A',
+                                u'#children': [
+                                    {
+                                        u'#type': u'vertical',
+                                        u'#ref': 'vert_1',
+                                        u'#children': [
+                                            {
+                                                u'#type': u'vertical',
+                                                u'#ref': u'vert_A11',
+                                                u'#children': [{u'#type': u'problem', u'#ref': u'prob_A1aa'}]
+                                            },
+                                        ]
+                                    },
+                                    {u'#type': u'vertical', u'#ref': 'vert_2', '#parents': [u'vert_A11']},
+                                ]
+                            },
+                            {
+                                u'#type': u'sequential',
+                                u'#ref': u'sub_B',
+                                u'#children': [
+                                    {u'#type': u'vertical', u'#ref': 'vert_3', '#parents': ['sub_A']},
+                                    {
+                                        u'#type': u'sequential',
+                                        u'#ref': 'sub_C',
+                                        '#parents': ['sub_A'],
+                                        u'#children': [{u'#type': u'problem', u'#ref': u'prob_BCb'}]
+                                    },
+                                ]
+                            },
+                        ]
+                    }
+                ]
+            }
+        ])
+
+    def test_collect_containing_subsection(self):
+        expected_containing_subsection = {
+            'course': 0,
+            'chapter': 0,
+            'sub_A': 1,
+            'sub_B': 1,
+            'sub_C': 3,
+            'vert_1': 1,
+            'vert_2': 1,
+            'vert_3': 2,
+            'vert_A11': 1,
+            'prob_A1aa': 1,
+            'prob_BCb': 3,
+        }
```

Comment:
Can we make the test even more precise by comparing the elements in the sets and not just the `count` of the elements in the set?
You can use the `#ref` names of the nodes to declare them in the expected dict.